crates/shrs_core/src/builtin/debug.rs

```diff
@@ -28,20 +31,25 @@ impl BuiltinCmd for DebugBuiltin {
         ctx: &mut Context,
         rt: &mut Runtime,
         args: &Vec<String>,
-    ) -> anyhow::Result<BuiltinStatus> {
+    ) -> anyhow::Result<CmdOutput> {
+        let mut out = String::new();
         let cli = Cli::try_parse_from(args)?;
 
         match &cli.command {
             None => {
-                println!("debug utility");
+                let s = "debug utility\n";
+                print!("{s}");
+                out += s;
             },
             Some(Commands::Env) => {
                 for (var, val) in rt.env.iter() {
-                    println!("{:?} = {:?}", var, val);
+                    let envs = format!("{:?} = {:?}\n", var, val);
+                    print!("{envs}");
+                    out += envs.as_str();
```

Comment:
i think that its a bit silly having to append to an output string. its also easy for users to use println by accident. i suggest perhaps to make some sort of wrapper class that helps with output. what we want in this case is a stringstream of some sort

maybe your command output writer can implement [BufWriter](https://doc.rust-lang.org/std/io/struct.BufWriter.html) trait. this definitely should use some more investigation